packages/jest-runtime/src/index.ts

```diff
@@ -1574,23 +1573,18 @@ export default class Runtime {
       return source;
     }
 
-    let transformedFile: TransformResult | undefined =
-      this._fileTransforms.get(filename);
-
-    if (transformedFile) {
-      return transformedFile.code;
-    }
-
-    transformedFile = this._scriptTransformer.transform(
+    const transformedFile = this._scriptTransformer.transform(
       filename,
       this._getFullTransformationOptions(options),
       source,
     );
 
-    this._fileTransforms.set(filename, {
-      ...transformedFile,
-      wrapperLength: this.constructModuleWrapperStart().length,
-    });
+    if (this._fileTransforms.get(filename)?.code !== transformedFile.code) {
```

Comment:
I think you can remove this `if` and just always populate the cache. Then it's a single `set` instead of a single `get` and _sometimes_ an additional `set`

Reply:
yeah that way we can prevent the additional set

Reply:
a `set` isn't more expensive than `get`